apps/accounts/views.py

```diff
@@ -154,14 +158,60 @@ def get(self, *args, **kwargs):
             )
             message = "Thanks, we've confirmed your email address. Now you can login with your username and password."
             messages.success(self.request, message)
-            return HttpResponseRedirect(
-                force_str(self.get_success_url(activated_user))
-            )
+            return HttpResponseRedirect(force_str(self.get_success_url(activated_user)))
 
         messages.error(self.request, error_message)
         return HttpResponseRedirect(force_str(self.get_success_url()))
 
 
+class DomainHashHomeView(LoginRequiredMixin, ListView):
+    """
+    Domain hash home page:
+    - used by external (non-staff) users to see a list of domain hashes they have created,
+    - renders the list of domain hashes a HTML template,
+    - requires the flag `domain_hash` enabled for the user (otherwise returns 404).
+    """
+
+    template_name = "provider_portal/domain_hash_index.html"
+    model = DomainHash
+
+    def get_queryset(self):
+        # TODO we want to allow users to see their own domain hashes, and
+
+        return self.model.objects.filter(
+            provider__in=[
+                prov.id
+                for prov in self.request.user.hosting_providers_explicit_perms.all()
+            ]
+        )
+
+    def get(self, *args, **kwargs):
+        self.object_list = self.get_queryset()
+        context = self.get_context_data()
+        import ipdb
+
+        return self.render_to_response(context)
+
+
+class DomainHashCreateView(LoginRequiredMixin, CreateView):
+    template_name = "provider_portal/domain_hash_new.html"
+    form_class = DomainHashForm
+    model = DomainHash
+
+
+class DomainHashDetailView(LoginRequiredMixin, DetailView):
+    template_name = "provider_portal/domain_hash_detail.html"
+    model = DomainHash
+
+    def get(self, request, *args, **kwargs):
```

Comment:
This is where I figure we might support a POST, or have a different endpoint to trigger the domain verification logic for a domain.

It's not implemented here, but we have it in the other PR